front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsControllerV2.groovy

```diff
@@ -0,0 +1,163 @@
+import com.netflix.spinnaker.front50.model.pipeline.PipelineDAO
+import com.netflix.spinnaker.front50.model.pipeline.PipelineStrategyDAO
+import com.netflix.spinnaker.front50.model.project.ProjectDAO
+import com.netflix.spinnaker.front50.validator.ApplicationValidator
+import groovy.util.logging.Slf4j
+import io.swagger.annotations.Api
+import io.swagger.annotations.ApiOperation
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.MessageSource
+import org.springframework.context.i18n.LocaleContextHolder
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.AccessDeniedException
+import org.springframework.security.access.prepost.PostFilter
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.Errors
+import org.springframework.validation.ObjectError
+import org.springframework.web.bind.annotation.*
+
+import javax.servlet.http.HttpServletResponse
+
+@Slf4j
+@RestController
+@RequestMapping("/v2/applications")
+@Api(value = "application", description = "Application API")
+public class ApplicationsControllerV2 {
+  @Autowired
+  MessageSource messageSource
+
+  @Autowired
+  ApplicationDAO applicationDAO
+
+  @Autowired
+  ProjectDAO projectDAO
+
+  @Autowired
+  NotificationDAO notificationDAO
+
+  @Autowired
+  PipelineDAO pipelineDAO
+
+  @Autowired
+  PipelineStrategyDAO pipelineStrategyDAO
+
+  @Autowired
+  List<ApplicationValidator> applicationValidators
+
+  @Autowired(required = false)
+  List<ApplicationEventListener> applicationEventListeners = []
+
+  @Autowired
+  Registry registry
+
+  @PreAuthorize("@fiatPermissionEvaluator.storeWholePermission()")
+  @PostFilter("hasPermission(filterObject.name, 'APPLICATION', 'READ')")
+  @ApiOperation(value = "", notes = """Fetch all applications.
+
+    Supports filtering by one or more attributes:
+    - ?email=my@email.com
+    - ?email=my@email.com&name=flex""")
+  @RequestMapping(method = RequestMethod.GET)
+  Set<Application> applications(@RequestParam Map<String, String> params) {
+    if (params.isEmpty()) {
+      return applicationDAO.all()
+    }
+
+    return applicationDAO.search(params)
+  }
+
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'CREATE')")
+  @ApiOperation(value = "", notes = "Create an application")
+  @RequestMapping(method = RequestMethod.POST, value = "")
+  Application create(@RequestBody final Application app) {
+    return getApplication().initialize(app).withName(app.getName()).save()
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'WRITE')")
+  @ApiOperation(value = "", notes = "Delete an application")
+  @RequestMapping(method = RequestMethod.DELETE, value = "/{application}")
+  void delete(@PathVariable String application, HttpServletResponse response) {
+    getApplication().initialize(new Application().withName(application)).delete()
+    response.setStatus(HttpStatus.ACCEPTED.value())
```

Comment:
Any thoughts from the gallery on status codes for deletes ... suppose it could either be 202 or 204. 

Personally, I'd be inclined to go with 204 since the delete has actually happened. 202 would imply it's *going* to happen.